packages/web-runtime/src/components/Topbar/UserMenu.vue

```diff
@@ -0,0 +1,174 @@
+          >
+            <avatar-image :width="32" :userid="userId" :user-name="user.displayname" />
+            <span class="profile-info-wrapper" :class="{ 'oc-py-xs': !user.email }">
+              <span v-text="user.displayname" />
+              <br v-if="user.email" />
+              <span v-if="user.email" class="oc-text-small" v-text="user.email" />
+            </span>
+          </oc-button>
+        </li>
+        <li v-for="(n, nid) in applicationsList" :key="`user-menu-${nid}`">
+          <oc-button v-if="n.url" type="a" appearance="raw" :target="n.target" :href="n.url">
+            <oc-icon :name="n.icon" class="oc-p-xs" />
+            <translate>{{ n.title }}</translate>
+          </oc-button>
+          <oc-button v-else type="router-link" appearance="raw" :to="{ path: n.path }">
+            <oc-icon :name="n.icon" class="oc-p-xs" />
+            <translate>{{ n.title }}</translate>
+          </oc-button>
+        </li>
+        <li>
+          <oc-button id="oc-topbar-account-logout" appearance="raw" @click="logout">
+            <oc-icon name="logout-box-r" fill-type="line" class="oc-p-xs" />
+            <translate>Log out</translate>
+          </oc-button>
+        </li>
+        <li v-if="quotaEnabled" class="storage-wrapper oc-pl-s">
+          <oc-icon name="cloud" fill-type="line" class="oc-p-xs" />
+          <div class="storage-wrapper-text">
+            <p class="oc-my-rm">
+              <span v-text="personalStorageLabel" />
+              <br />
+              <span class="oc-text-small" v-text="personalStorageDetailsLabel" />
+            </p>
+            <oc-progress
+              :value="parseInt(quotaUsagePercent)"
+              :max="100"
+              size="small"
+              :variation="quotaProgressVariant"
+            />
+          </div>
+        </li>
+      </oc-list>
+    </oc-drop>
+  </nav>
+</template>
+
+<script>
+import { mapGetters } from 'vuex'
+import filesize from 'filesize'
+
+export default {
+  props: {
+    applicationsList: {
+      type: Array,
+      required: false,
+      default: () => []
+    }
+  },
+  computed: {
+    ...mapGetters(['quota', 'user']),
+
+    userId() {
+      return this.user.username || this.user.id
+    },
+    personalStorageLabel() {
+      return this.$gettextInterpolate('Personal storage (%{percentage}% full)', {
+        percentage: this.quotaUsagePercent
+      })
+    },
+    personalStorageDetailsLabel() {
+      return this.$gettextInterpolate('%{used} of %{total} used', {
+        used: this.quotaUsed,
+        total: this.quotaTotal
+      })
+    },
+    quotaEnabled() {
+      return !!this.quota
+    },
+    quotaTotal() {
+      return filesize(this.quota.total)
+    },
+    quotaUsed() {
+      return filesize(this.quota.used)
+    },
+    quotaUsagePercent() {
+      return ((this.quota.used / this.quota.total) * 100).toFixed(2)
+    },
+    quotaProgressVariant() {
+      if (this.quotaUsagePercent < 80) return 'primary'
```

Comment:
would it make sense to have the thresholds as an settings value? I am not sure if they are always the same

Reply:
Product question for @tbsbdr, sounds like a lot of complexity for something no one asked for so far:)